client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -442,7 +442,10 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 			verticalId={ siteVerticalId }
 			locale={ locale }
 			onSelect={ pickDesign }
-			onPreview={ previewDesign }
+			onPreview={ ( design: Design, variation?: StyleVariation ) => {
+				setSelectedStyleVariation( variation );
```

Comment:
What if we put this logic directly inside the `previewDesign()` function?

https://github.com/Automattic/wp-calypso/blob/c0b28f88134d15f7c15b72fdf0006a4499ac012a/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx#L139-L143

We can change the second parameter to match with what you have here (`variation?: StyleVariation`), removing `positionIndex`. I tried searching for `positionIndex` usage but it looks like it is never used? Or was it a mistake when we developed UDP? 😅

Reply:
It's used to track the positioning of the selected generated design. I think we can change the second argument to an object, so that we can send data as `previewDesign( design, { variation } )` or `previewDesign( design, { positionIndex } )`, what do you think?

Reply:
Forgot to link the section where `positionIndex` is [used](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx#L237).

Reply:
There is no call to `previewDesign()` that passes `positionIndex`. Or am I missing something?

Reply:
Oh I see what you mean, I think that's just copied over from the v13n design picker. Yup, in UDP there's no call that passes `positionIndex`

Reply:
Updated 🙂